src/StatamicNav.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Rapidez\Statamic;
+
+use Illuminate\Support\Facades\Cache;
+use Statamic\Statamic;
+use Statamic\Eloquent\Entries\Entry;
+use Rapidez\Core\Facades\Rapidez;
+use Statamic\Structures\Page;
+
+class StatamicNav
+{
+    public function nav(string $tag): array
+    {
+        if (!str($tag)->startsWith('nav:')) {
+            return [];
+        }
+
+        return Cache::rememberForever($tag . '-' . config('rapidez.store'), fn() => $this->buildMenu($tag));
+    }
+
+    protected function buildMenu(string $key): array
+    {
+        $nav = Statamic::tag($key)->fetch();
+
+        return $this->buildTree($nav);
+    }
+
+    private function buildTree(array $items, $parentId = null): array
+    {
+        $branch = [];
+
+        foreach ($items as $item) {
+            if ($item['parent'] == $parentId) {
+                $children = $this->buildTree($item['children'], $item['id']);
+                if ($children) {
+                    $element['children'] = $children;
+                }
+            }
+
+            $item['url'] = $this->determineEntryUrl($item['entry_id']->augmentable());
+            $branch[] = $item;
+        }
+
+        return $branch;
+    }
+
+    public function determineEntryUrl(Entry|Page $entry, string $nav = 'global-link'): string
+    {
+        return Cache::rememberForever(sprintf('%s-%s-nav-%s', config('rapidez.store'), $nav, $entry->id()), function () use ($entry) {
```

Comment:
I'm afraid we'll be creating a lot of caches, and retrieving a lot of caches by storing and retrieving this way.

We can reduce it by saving all urls into an array per `store-nav`
like done here
https://github.com/rapidez/core/pull/567/files#diff-061a7aa858d94ff473663e973e3097fbf1944640542b81be93f1961b42604bb3

however here we have a singleton and thus access to do 
`$this->navCache[$nav][$entry->id()] = $value`
instead of having to use both the array driver and regular cache driver